src/VisualStudio/CSharp/Impl/EventHookup/EventHookupSessionManager_EventHookupSession.cs

```diff
@@ -112,17 +112,18 @@ public void Cancel()
                         _cancellationTokenSource.Token,
                         TaskScheduler.Default);
 
-                    var continuedTask = this.GetEventNameTask.SafeContinueWith(t =>
+                    var continuedTask = this.GetEventNameTask.SafeContinueWithFromAsync(
+                        async t =>
                         {
-                            AssertIsForeground();
+                            await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(_cancellationTokenSource.Token);
```

Comment:
❕ This needs to yield if called from the main thread in order to preserve the original behavior.